src/scripts/pokemons/PokemonHelper.ts

```diff
@@ -437,7 +437,7 @@ class PokemonHelper extends TmpPokemonHelper {
         return locations[PokemonLocationType.Dungeon] ||
             locations[PokemonLocationType.DungeonBoss] ||
             locations[PokemonLocationType.DungeonChest] ||
-            locations[PokemonLocationType.Evolution] ||
+            (locations[PokemonLocationType.Evolution] && PokemonHelper.getPokemonStonePrevolution(pokemonName)) || // Only stone evolutions gives EVs
```

Comment:
This searches through the pokemonList looking for a stone prevo, but we should already have the info on locations[PokemonLocationType.Evolution] to use
```suggestion
            (locations[PokemonLocationType.Evolution] as EvoData[])?.some((evo) => evo.trigger === EvoTrigger.STONE) || // Only stone evolutions gives EVs
```

Reply:
That works and is more effective. Thanks!